app/services/support_interface/qualifications_export.rb

```diff
@@ -0,0 +1,144 @@
+        degrees = degrees(qualifications).sort_by(&:subject)
+
+        output = {
+          'Candidate id' => application_form.candidate_id,
+          'Support ref' => application_form.support_reference,
+          'Phase' => application_form.phase,
+          'Cycle' => application_form.recruitment_cycle_year,
+
+          'Outcome (offer, rejected etc.)' => application_choice.status,
+          'Reason for Rejection' => application_choice.rejection_reason,
+          'Course Code' => course.code,
+          'Provider Code' => course.provider.code,
+
+          'GCSE maths grade' => maths_gcse_grade(qualifications),
+
+          'GCSE science single grade' => science_single_gcse_grade(qualifications),
+          'GCSE science double grade' => science_double_gcse_grade(qualifications),
+          'GCSE science triple grade' => science_triple_gcse_grade(qualifications),
+
+          'GCSE English unstructured grade' => english_unstructured_gcse_grade(qualifications),
+          'GCSE English single grade' => english_structured_gcse_grades(qualifications, 'english_single_award'),
+          'GCSE English double grade' => english_structured_gcse_grades(qualifications, 'english_double_award'),
+          'GCSE English language grade' => english_structured_gcse_grades(qualifications, 'english_language'),
+          'GCSE English literature grade' => english_structured_gcse_grades(qualifications, 'english_literature'),
+          'GCSE English studies single grade' => english_structured_gcse_grades(qualifications, 'english_studies_single_award'),
+          'GCSE English studies double grade' => english_structured_gcse_grades(qualifications, 'english_studies_double_award'),
+          'GCSE English other grade' => english_other_gcse_grade(qualifications),
+
+          'A level 1 subject' => a_levels[0].try(:subject),
+          'A level 1 grade' => a_levels[0].try(:grade),
+          'A level 2 subject' => a_levels[1].try(:subject),
+          'A level 2 grade' => a_levels[1].try(:grade),
+          'A level 3 subject' => a_levels[2].try(:subject),
+          'A level 3 grade' => a_levels[2].try(:grade),
+          'A level 4 subject' => a_levels[3].try(:subject),
+          'A level 4 grade' => a_levels[3].try(:grade),
+          'A level 5 subject' => a_levels[4].try(:subject),
+          'A level 5 grade' => a_levels[4].try(:grade),
+
+          'Degree 1 type' => degrees[0].try(:qualification_type),
+          'Degree 1 grade' => degrees[0].try(:grade),
+          'Degree 2 type' => degrees[1].try(:qualification_type),
+          'Degree 2 grade' => degrees[1].try(:grade),
+
+          'Number of other qualifications provided' => other_qualification_count(qualifications),
+        }
+        output
+      end
+    end
+
+  private
+
+    def maths_gcse_grade(qualifications)
+      maths_gcse = qualifications.where(level: :gcse, subject: :maths).first
+      maths_gcse.try(:grade)
+    end
+
+    def science_single_gcse_grade(qualifications)
```

Comment:
Do you think we need separate columns for each of these? 

Could we just have:

def science_single_gcse_grade(application_form)
  science_gcse = application_form.science_gcse
  science_gcse.structured_grade || science_gcse.unstructured_grade
end

Depending on the returned grades it should be fairly explicit whether it was a single, double or triple award. Might be worth asking Mike.

Reply:
Although your approach is far more filterable. So it may well be more appropriate.

Reply:
We spoke to Mike quite a bit when we built this and he said that having the sciences in separate columns is really helpful.